### ISSUE TITLE
Catalog Products List widget: Sku chooser grid not showing when editing an existing condition for the attribute SKU

### ISSUE THREAD

**Issue by joost-florijn-kega** [magento/magento2#6985]
### Preconditions
1. branch develop HEAD
2. PHP 7
### Steps to reproduce
1. Create a new 'Catalog products widget' with a condition for attribute SKU.
2. Save and Reopen the widget.
3. Change the select SKU's by opening the sku chooser grid.
### Expected result
1. Visible sku chooser grid with the selected SKU's preselected.
### Actual result
1. The loader mask is removed but no sku chooser grid is visible.
   ![image](https://cloud.githubusercontent.com/assets/16535591/19343399/59b7bf5c-9136-11e6-87a9-b43373bc7f97.png)

The result of the ajax call to load the sku chooser grid contains invalid javascript.  
https://example.com/admin/catalog_rule/promo_widget/chooser/attribute/sku/key/2bccdf1408a5084e5a0f20e0eb47875fb87baa71fd3b70cfeec8a99ad17a537f/?isAjax=true

```
<script>
    var deps = [];
    deps.push('mage/adminhtml/grid');

    require(deps, function(){

        //<![CDATA[

    skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject = new varienGrid("skuChooserGrid_id_19ad8c88729df84057c3dc311a784312", 'https://euronics.dev/sitemanager/catalog_rule/promo_widget/chooser/current_grid_id/skuChooserGrid_id_19ad8c88729df84057c3dc311a784312/attribute/sku/key/2bccdf1408a5084e5a0f20e0eb47875fb87baa71fd3b70cfeec8a99ad17a537f/?isAjax=true', 'page', 'sort', 'dir', 'filter');
    skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject.useAjax = '1';
        skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject.rowClickCallback = .chooserGridRowClick.bind();
            skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject.checkboxCheckCallback = .chooserGridCheckboxCheck.bind();
            skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject.initRowCallback = .chooserGridRowInit.bind();
    skuChooserGrid_id_19ad8c88729df84057c3dc311a784312JsObject.initGridRows();

        //]]>

});
</script>
```

It is missing the javascript reference to the form object which is set in the block [Chooser/SKU](https://github.com/magento/magento2/blob/develop/app/code/Magento/CatalogRule/Block/Adminhtml/Promo/Widget/Chooser/Sku.php#L74).

When adding a new condition rule the following url is used:
https://example.com/admin/catalog_widget/product_widget/conditions/form/options_fieldset67a77e971a7c331b6eaefcaf2f596097/key/a6de72a5c50a80fd1db084dbf78ca3fb18056864ac4ca8ca1883e27e09c300f4/?isAjax=true

You can see that the javascript form object reference is send as param form. Because of this the sku chooser grid is shown for new condition rules.

**Comment by dthampy** [comment]
MAGETWO-60613 is created internally to track this issue.

**Comment by magento-engcom-team** [comment]
@joost-florijn-kega, thank you for your report.
The issue is already fixed in 2.2.0
